protocol.md

```diff
@@ -0,0 +1,752 @@
+is a "call req", and all subsequent frames are "call req continue" frames.
+
+### flags:1
+
+Used to control fragmentation. Valid flags:
+
+`0x01` | more fragments follow
+
+If the fragments flag isn't set, then this is the only/last frame for this
+message id.
+
+### ttl:4
+
+Time To Live in milliseconds. Intermediaries should decrement this as
+appropriate when making dependent requests. Since all numbers are unsigned the
+ttl can never be less than 0. Care should be taken when decrementing ttl to
+make sure it doesn't go below 0. Requests should never be sent with ttl of 0.
+If the ttl expires an error response should be generated.
+
+### tracing:24
+
+Tracing payload in zipkin format. The contents are:
+
+`spanid:8`   | int64 that identifies the current *span*
+`parentid:8` | int64 of the previous *span*
+`traceid:8`  | int64 assigned by the original requestor
+
+A *span* is a logical operation like call or forward
+
+The `traceid` does not change as it propagates through various services.
+
+When a request enters our system the edge-most requester selects a `traceid`
+which will remain unchanged as this message and any dependent messages move
+through the system. Each time a new request is generated a new `spanid` is
+generated. If it was started from a previous request that id is moved to the
+`parentid`.
+
+### traceflags:1
+
+`0x01` | tracing enabled for this request
+
+When this flag is not set, the tracing data is still required to be present in
+the frame, but the tracing data will not necessarily be sent to zipkin.
+
+### service~2
+
+UTF-8 string identifying the destination service to which this request should be
+routed.
+
+### nh:1 (hk~1 hv~1){nh}
+
+Transport headers described below in the "Transport Headers" section.
+
+### csumtype:1 (csum:4){0,1}
+
+Checksum described below in the "checksums" section.
+
+### arg1~2 arg2~2 arg3~2
+
+The meaning of the three args depends on the systems on each end. The format of
+arg1, arg2, and arg3 is unspecified at the transport level. These are opaque
+binary blobs as far as tchannel is concerned.
+
+Future versions will likely allow callers to specify specific service instances
+on which to run this request, or a mechanism to route a certain percentage of
+all traffic to a subset of instances for canary analysis.
+
+## Payload: call res (0x04)
+
+Schema: `flags:1 code:1 tracing:24 traceflags:1 nh:1 (hk~1 hv~1){nh} csumtype:1 (csum:4){0,1} arg1~2 arg2~2 arg3~2`
+
+Very similar to call res (type 0x03), differing only in:
+
+- adds a `code` field
+- no `ttl` field
+- no `service` field
+
+All common fields have identical definition to call res, see its section above for detail.
+
+Headers described below in the "Transport Headers" section.
+
+### code:1
+
+Response code:
+
+code | name  | description
+-----|-------|------------
+0x00 | OK    | everything is great and we value your contribution.
+0x01 | Error | application error, details are in the args.
```

Comment:
Nit: Inconsistent with tables above where items in the `code` column were fixed-width.